go-controller/pkg/kubevirt/external_ids.go

```diff
@@ -0,0 +1,36 @@
+package kubevirt
+
+import (
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	ktypes "k8s.io/apimachinery/pkg/types"
+)
+
+func ExtractVMFromExternalIDs(externalIDs map[string]string) *ktypes.NamespacedName {
+	namespace, ok := externalIDs[string(libovsdbops.NamespaceIndex)]
+	if !ok {
+		return nil
+	}
+	vmName, ok := externalIDs[string(libovsdbops.VirtualMachineIndex)]
+	if !ok {
+		return nil
+	}
+	return &ktypes.NamespacedName{Namespace: namespace, Name: vmName}
+}
+
+// OwnsItAndIsOrphanOrWrongZone return true if kubevirt owns this this OVN NB
+// resource by check if it has the VM name external id and also check if the
```

Comment:
by checking if it has the VM name in external_ids and also checks if the expected ovn zone corresponds with the one it created via the OvnZoneExternalIDKey